### PR TITLE
#162163265 send user an email with link to reset password

### PR DESCRIPTION
**Pull request by joshtrigger**
**What does this PR do?**
send user email with password reset link upon user's request

**Description of the task**
when a user has forgots password they request for password reset. user provides email address and then email with the link is sent.

**How should be manually tested?**
clone the repository and run `pip install -r requirements.txt`
Goto the `/api/password_reset/`
Expected output:
```
{
    "message":"An email has been sent to this email"
}
```
Go to your email and click on the link that has been sent
This will run on this endpoint `/api/password_reset_confirm/<slug>/`
This is what you should receive
```
{
		"password":"qwerty123",
		"confirm_password":"qwerty123"
}
```
Expected output
```
{
    "message": {
        "detail": [
            "Password successfully reset"
        ]
    }
}
```
**NOTE:** Only signed up users can use this service

**What are the relevant pivotal tracker stories?**
#13

### CODE REVIEW

**Review by collin5 — CHANGES_REQUESTED** [review]
Please fix failing tests

File: authors/settings.py [hunk 1]
```diff
@@ -152,4 +152,11 @@
...
 # Activate Django-Heroku; must be at the very bottom
-django_heroku.settings(locals())
\ No newline at end of file
+django_heroku.settings(locals())
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_HOST_USER = os.getenv('EMAIL', '')
+EMAIL_HOST_PASSWORD = os.getenv('EMAIL_PASSWORD', '')
+EMAIL_PORT = 587
```
Comment (collin5) [inline authors/settings.py]: You should get all these from environment before setting defaults.

**Review by collin5 — CHANGES_REQUESTED** [review]

File: authors/settings.py (hunk 1 above)
Comment (joshtrigger) [reply authors/settings.py]: fixed issue

Comment (joshtrigger) [reply authors/settings.py]: plus the failing tests

**Review by bibangamba — APPROVED** [review]

File: authors/settings.py [hunk 2]
```diff
@@ -163,4 +163,11 @@
...
-django_heroku.settings(locals())
\ No newline at end of file
+django_heroku.settings(locals())
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_HOST_USER = os.getenv('EMAIL', '')
+EMAIL_HOST_PASSWORD = os.getenv('EMAIL_PASSWORD', '')
+EMAIL_PORT = os.getenv('EMAIL_PORT', '')
+EMAIL_USE_TLS = os.getenv('TLS', '')
```
Comment (collin5) [inline authors/settings.py]: `os.getenv('TLS', True)`

File: authors/settings.py [hunk 3]
```diff
@@ -163,4 +163,11 @@
...
 # Activate Django-Heroku; must be at the very bottom
-django_heroku.settings(locals())
\ No newline at end of file
+django_heroku.settings(locals())
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_HOST_USER = os.getenv('EMAIL', '')
+EMAIL_HOST_PASSWORD = os.getenv('EMAIL_PASSWORD', '')
+EMAIL_PORT = os.getenv('EMAIL_PORT', '')
```
Comment (collin5) [inline authors/settings.py]: `os.getenv('EMAIL_PORT', 587)`

**Review by collin5 — CHANGES_REQUESTED** [review]
One small change before we merge. Thanks!

**Review by KicaRonaldOkello — APPROVED** [review]

**Review by collin5 — APPROVED** [review]

**Pull request merged**